src/sedpack/io/tfrec/read.py

```diff
@@ -33,15 +33,15 @@ class IterateShardTFRec(IterateShardBase[T]):
     """
 
     def __init__(
-            self,
-            dataset_structure: DatasetStructure,
-            process_record: Callable[[ExampleT], Any] | None,
-            num_parallel_calls: int = os.cpu_count(),
+        self,
+        dataset_structure: DatasetStructure,
+        process_record: Callable[[ExampleT], Any] | None,
+        num_parallel_calls: int = os.cpu_count() or 4,
```

Comment:
Does that solves issue #8 or do we also need an upper bound?

Reply:
No. Issue #8 depends on proper benchmarking.

Reply:
Ack